dataloader/dataloader.hpp

```diff
@@ -0,0 +1,225 @@
+/**
+ * @file dataloader.hpp
+ * @author Kartik Dutt
+ * 
+ * Definition of Dataloader for popular datasets.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core/data/scaler_methods/min_max_scaler.hpp>
+#include <mlpack/core/math/shuffle_data.hpp>
+#include <mlpack/core/data/split_data.hpp>
+#include <mlpack/prereqs.hpp>
+#include <dataloader/datasets.hpp>
+#include <mlpack/core.hpp>
+#include <utils/utils.hpp>
+
+
+#ifndef MODELS_DATALOADER_HPP
+#define MODELS_DATALOADER_HPP
+
+/**
+ * Dataloader class to load popular datasets.
+ *
+ * @tparam DatasetX Datatype for loading input features.
+ * @tparam DatasetY Datatype for prediction features.
+ * @tparam ScalerType mlpack's Scaler Object for scaling features.
+ */
+template<
+  typename DatasetX = arma::mat,
+  typename DatasetY = arma::mat,
+  class ScalerType = mlpack::data::MinMaxScaler
+>
+class DataLoader
+{
+ public:
+  //! Create DataLoader object.
+  DataLoader();
+
+  /**
+   * Constructor for DataLoader. This is used for loading popular Datasets such as
+   * MNIST, ImageNet, Pascal VOC and many more.
+   * 
+   * @param datasetPath Path or name of dataset.
+   * @param shuffle whether or not to shuffle the data.
+   * @param ratio Ratio for train-test split.
+   * @param useScaler Use feature scaler for pre-processing the dataset.
+   * @param augmentation Adds augmentation to training data only.
+   * @param augmentationProbability Probability of applying augmentation on dataset.
+   */
+  DataLoader(const std::string& dataset,
```

Comment:
Do you think we should provide a Constructor that returns the loaded and processed dataset as well?

Reply:
I am terribly sorry but I didn't get your point here. I assuming it means that we should add a constructor for custom dataset as well ?

Reply:
Nevermind I thought I have to first create the `DataLoader` object and after that use `DownloadDataset` to get the dataset, but I can just use `TrainX`. Sorry for the confusion.

Reply:
Great, making rest of the changes now. Thanks a ton!